sovtokenfees/sovtokenfees/static_fee_req_handler.py

```diff
@@ -69,11 +66,26 @@ def __init__(self, ledger, state, token_ledger, token_state, utxo_cache,
         # paying sovtokenfees
         self.uncommitted_state_roots_for_batches = []
 
-    @staticmethod
-    def has_fees(request) -> bool:
-        return hasattr(request, FEES) and isinstance(request.fees, list) \
-               and len(request.fees) > 0 and isinstance(request.fees[0], list) \
-               and len(request.fees[0]) > 0
+    def has_fees(self, request) -> bool:
+        try:
+            self.validate_fees(request)
+            return True
+        except InvalidClientMessageException:
+            return False
+
+    def validate_fees(self, request):
+        has_fees = hasattr(request, FEES)
+        if has_fees:
+            fees = request.fees
+            error = self._txn_fees_validator.validate(fees)
```

Comment:
This validation is a little bit of an overkill. This field is already validated by node. Here is how it is done:
* Plugin defines a dict of its special fields that are inserted to the root object of request: https://github.com/sovrin-foundation/token-plugin/blob/2c1e378381326d06253d21b7b645f383b25f402d/sovtokenfees/sovtokenfees/__init__.py#L7
* Plenum collects all these fields to global dict called PLUGIN_CLIENT_REQUEST_FIELDS: https://github.com/hyperledger/indy-plenum/blob/f4a350a09e72512863049760809d47945f14270a/plenum/__init__.py#L63
* Here they are added to the validator of requests: https://github.com/hyperledger/indy-plenum/blob/decd12512afc997960c7cb8fd2dc12bf8f7e8a17/plenum/common/messages/client_request.py#L119
* And then they are validated with all other fields here: https://github.com/hyperledger/indy-plenum/blob/a8f678a1bf705b4b0eb4767586bff5d3d6094ab4/plenum/common/messages/message_base.py#L23

So there is no need to make this validation once again.